api/v1alpha1/clusterextension_types.go

```diff
@@ -73,8 +73,32 @@ type ClusterExtensionSpec struct {
 	// the bundle may contain resources that are cluster-scoped or that are
 	// installed in a different namespace. This namespace is expected to exist.
 	InstallNamespace string `json:"installNamespace"`
+
+	//+kubebuilder:Optional
+	// Preflight defines the strategy for the preflight checks (i.e. as of now, the CRD upgrade safety checks) to be applied or skipped when attempting to install the cluster extension.
+	Preflight PreflightConfig `json:"preflight,omitempty"`
 }
 
+// PreflightConfig holds the config for the preflight checks. Currently, this is implemented for the
+// CRDUpgradeSafety preflight check and can be extended in the future to add other preflight checks.
+type PreflightConfig struct {
+	CRDUpgradeSafety CRDUpgradeSafetyPreflightConfig `json:"crdUpgradeSafety,omitempty"`
+}
+
+// CRDUpgradeSafetyPreflightConfig is a custom type to hold value for the CRDUpgradeSafetyPreflightConfig
+// that enables or disables the CRD Upgrade Safety validations.
+type CRDUpgradeSafetyPreflightConfig string
```

Comment:
We may want to make this a struct so we can expand upon it in the future by adding additional optional fields. I'm fine with the enum approach over a boolean and it can be the first field in this new struct

Reply:
Agreed 
Updated in [commit](https://github.com/operator-framework/operator-controller/pull/845/commits/e35ad160f1dbd3459bcebf0a20c0f109b49f4982)